### PR TITLE
Fix sensitivity op

### PR DESCRIPTION
**Pull request by JakobAsslaender**
Hi, 
I mostly generalized the input to abstract types as some operators in this package return type `subArray` and hence do not work well with this operator. 

Meanwhile.... I also fixed a bug for a missing `storage_type` function for `DiagOp` and I also capitalized all constructors of the latter type. The latter is a breaking change, but I found it kinda confusing (and was confused by it) that the main constructor and the convenience constructors had different capitalizations. 

Sorry for putting multiple things in on PR -- that was me being lazy.

### CODE REVIEW

File: MRIOperators/src/MRIOperators.jl [hunk 1]
```diff
@@ -50,21 +50,21 @@ function vcat(A::AbstractLinearOperator, n::Int)
   return op
 end
 
-function diagOpProd(y::AbstractVector{T}, x::AbstractVector{T}, nrow::Int, xIdx, yIdx, ops :: AbstractLinearOperator...) where T
```
Comment (tknopp) [inline MRIOperators/src/MRIOperators.jl:53]: this should not be capitalized. It is not a constructor.

File: MRIOperators/src/MRIOperators.jl [hunk 2]
```diff
@@ -50,21 +50,21 @@ function vcat(A::AbstractLinearOperator, n::Int)
...
+function DiagOpProd(y::AbstractVector{T}, x::AbstractVector{T}, nrow::Int, xIdx, yIdx, ops :: AbstractLinearOperator...) where T
   @floop for i=1:length(ops)
     mul!(view(y,yIdx[i]:yIdx[i+1]-1), ops[i], view(x,xIdx[i]:xIdx[i+1]-1))
   end
   return y
 end
 
-function diagOpTProd(y::AbstractVector{T}, x::AbstractVector{T}, ncol::Int, xIdx, yIdx, ops :: AbstractLinearOperator...) where T
```
Comment (tknopp) [inline MRIOperators/src/MRIOperators.jl:60]: this as well

File: MRIOperators/src/MRIOperators.jl [hunk 3]
```diff
@@ -50,21 +50,21 @@ function vcat(A::AbstractLinearOperator, n::Int)
...
+function DiagOpTProd(y::AbstractVector{T}, x::AbstractVector{T}, ncol::Int, xIdx, yIdx, ops :: AbstractLinearOperator...) where T
   @floop for i=1:length(ops)
     mul!(view(y,yIdx[i]:yIdx[i+1]-1), transpose(ops[i]), view(x,xIdx[i]:xIdx[i+1]-1))
   end
   return y
 end
 
-function diagOpCTProd(y::AbstractVector{T}, x::AbstractVector{T}, ncol::Int, xIdx, yIdx, ops :: AbstractLinearOperator...) where T
```
Comment (tknopp) [inline MRIOperators/src/MRIOperators.jl:67]: and this

File: MRIOperators/src/MRIOperators.jl [hunk 4]
```diff
@@ -98,12 +98,15 @@ mutable struct DiagOp{T} <: AbstractLinearOperator{T}
 end
 
 
+LinearOperators.storage_type(op::DiagOp) = typeof(op.Mv5)
+
+
 """
-    diagOp(ops :: AbstractLinearOperator...)
+    DiagOp(ops :: AbstractLinearOperator...)
```
Comment (tknopp) [inline MRIOperators/src/MRIOperators.jl:105]: this change is ok!

File: MRIOperators/src/MRIOperators.jl [hunk 5]
```diff
@@ -117,16 +120,16 @@ function diagOp(ops :: AbstractLinearOperator...)
   yIdx = cumsum(vcat(1,[ops[i].nrow for i=1:length(ops)]))
 
   Op = DiagOp{S}( nrow, ncol, false, false,
-                     (res,x) -> (diagOpProd(res,x,nrow,xIdx,yIdx,ops...)),
-                     (res,y) -> (diagOpTProd(res,y,ncol,yIdx,xIdx,ops...)),
-                     (res,y) -> (diagOpCTProd(res,y,ncol,yIdx,xIdx,ops...)), 
+                     (res,x) -> (DiagOpProd(res,x,nrow,xIdx,yIdx,ops...)),
```
Comment (tknopp) [inline MRIOperators/src/MRIOperators.jl:123]: these need to be changed accordingly

**Comment by JakobAsslaender** [comment]
Fixed the capitalization as suggested.

**Pull request merged**